cmd/clusterctl/client/tree/discovery.go

```diff
@@ -280,6 +287,40 @@ func addMachinePoolsToObjectTree(ctx context.Context, c client.Client, namespace
 	}
 }
 
+func addMachineHealthChecksToObjectTree(ctx context.Context, c client.Client, cluster *clusterv1.Cluster, tree *ObjectTree) {
+	if mhcList, err := getMachineHealthChecksInCluster(ctx, c, cluster.Namespace, cluster.Name); err == nil {
```

Comment:
```suggestion
	if mhcList, err := getMachineHealthChecksInCluster(ctx, c, cluster.Namespace, cluster.Name); err == nil && len(mhcList.Items) > 0 {
```

If there are no MHCs associated with the cluster it just shows an empty `MachineHealthChecks` group in the output with nothing under it. The output looks a little weird in such cases.